### ISSUE TITLE
Investigate using `strings: :copy` to reduce memory pressure

### ISSUE THREAD

**Issue by axelson** [elixir-lsp/elixir-ls#236]
Investigate using `strings: :copy` option to `Jason.decode!(body)` in `PacketStream`

Original discussion in #169

**Comment by wingyplus** [comment]
I'm not familiar with profiling with Elixir so much. I can help profiling it if you have some guidance. :)

**Comment by lukaszsamson** [comment]
I did some tests. The workload was editing large files and opening many files as I tried to hit the case then references to binary parts of big messages are kept in process memory. I wasn't able tot trigger memory leaks (at least on macOS in OTP 23). Binaries are only a small fraction of total used memory - Dialyzer process tends to be the biggest user. In fact `:copy` slightly increased memory usage.

`strings: :reference`
<img width="241" alt="Screenshot 2020-12-06 at 12 44 55" src="https://user-images.githubusercontent.com/1078186/101279288-8389e400-37c1-11eb-81f6-ea4e6e49951a.png">

`strings: :copy`
<img width="220" alt="Screenshot 2020-12-06 at 12 36 29" src="https://user-images.githubusercontent.com/1078186/101279293-8be21f00-37c1-11eb-86e4-d3ddde7f5a8e.png">
